main.py

```
@@ -31,43 +31,30 @@ def main() -> None:
-
     # Fill the translated text into XML structure
     print("→ Calling Filler.fill()...")
     try:
-        result = filler.fill(
-            source_ele=source_ele,
-            translated_text=translated_text,
-            on_fail=lambda err: print(f"  ✗ Validation error: {err}"),
-        )
+        translated_ele = translator.translate(list(source_ele))
+        if translated_ele is None:
+            raise RuntimeError("Filler returned None as the result XML element")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)